packages/kit/types/index.d.ts

```diff
@@ -1188,7 +1204,7 @@ export function text(body: string, init?: ResponseInit): Response;
  * @param status The [HTTP status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses). Must be in the range 400-599.
  * @param data Data associated with the failure (e.g. validation errors)
  */
-export function fail<T extends Record<string, unknown> | undefined>(
+export function fail<T extends Record<string, unknown> | undefined = undefined>(
```

Comment:
I think it makes sense to set the default value to `undefined`. If you don't pass any `data`, the resulting type should be `undefined` and not `Record<string, unknown>`. This change should not break anything. When you pass any data to the function, the generic will be inferred automatically.